src/wowman/toc.clj

```diff
@@ -98,40 +98,63 @@
+               :dirname dirname
+               :label label
+               :description (or (:notes keyvals) (:description keyvals)) ;; :notes is preferred but we'll fall back to :description
+               :interface-version (or (some-> keyvals :interface utils/to-int)
+                                      default-interface-version)
+               :installed-version (:version keyvals)
+
+               ;; toc file dependency values describe *load order*, not *packaging*
+               ;;:dependencies (:dependencies keyvals)
+               ;;:optional-dependencies (:optionaldependencies keyvals)
+               ;;:required-dependencies (:requireddeps keyvals)
+               }]
+    (merge alias addon nfo-contents)))
+
+(defn-spec parse-addon-toc-guard (s/or :ok ::sp/toc, :error nil?)
+  "wraps the `parse-addon-toc` function and ensures no unhandled exceptions cause a cascading failure"
   [addon-dir ::sp/extant-dir]
-  (if-let [keyvals (read-addon-dir addon-dir)]
-    (let [;;decoded-title (:title keyvals) ;; TODO: title may be encoded: https://wow.gamepedia.com/UI_escape_sequences
-          dirname (fs/base-name addon-dir) ;; /foo/bar/baz => baz
-          install-dir (str (fs/parent addon-dir)) ;; /foo/bar/baz => /foo/bar
-          nfo-contents (nfo/read-nfo install-dir dirname)
-          label (:title keyvals)
-          alias (when (contains? aliases label) ;; ensure :alias is absent if not present
-                  {:alias (get aliases label)})
-          addon {:name (normalise-name (:title keyvals))
-                 :dirname dirname
-                 :label label
-                 :description (or (:notes keyvals) (:description keyvals)) ;; :notes is preferred but we'll fall back to :description
-                 :interface-version (-> keyvals :interface utils/to-int)
-                 :installed-version (:version keyvals)
-
-                 ;; toc file dependency values describe *load order*, not *packaging*
-                 ;;:dependencies (:dependencies keyvals)
-                 ;;:optional-dependencies (:optionaldependencies keyvals)
-                 ;;:required-dependencies (:requireddeps keyvals)
-                 }]
-      (merge alias addon nfo-contents))
-
-    ;; ignore failures to parse Blizzard addons
-    (when-not (.startsWith (fs/base-name addon-dir) "Blizzard_")
-      (warn "failed to find .toc file:" addon-dir))))
+  (try
+    (if-let [keyvals (read-addon-dir addon-dir)]
+      ;; we found a .toc file, now parse it
+      (parse-addon-toc addon-dir keyvals)
+      ;; we didn't find a .toc file, but just ignore it if it looks like an official addon dir
+      (when-not (.startsWith (fs/base-name addon-dir) "Blizzard_")
+        ;; we didn't find a .toc file and it really should have one. warn the user
+        (warn "failed to find .toc file:" addon-dir)))
+    (catch Exception e
+      ;; this addon really messed us up. don't propagate the failure, just report it and return nil
```

Comment:
"this addon failed some how. don't propagate the exception, just report it and return nil"